tree.go

```diff
@@ -78,8 +79,9 @@ func (c *CompressedMultiproof) Decompress() *Multiproof {
 // Node represents a node in the tree
 // backing of a SSZ object.
 type Node struct {
-	left  *Node
-	right *Node
+	left    *Node
+	right   *Node
+	isEmpty bool
```

Comment:
Adds a flag so we know the zero order hash is actually an empty node and we shouldn't traverse the tree further for the right node.